src/operations.rs

```diff
@@ -136,3 +135,67 @@ pub fn garbage_collect_versions(config_data: &mut JuliaupConfig) -> Result<()> {
 
     Ok(())
 }
+
+fn _remove_symlink(
+    symlink_path: &Path,
+) -> Result<()> {
+    std::fs::create_dir_all(symlink_path.parent().unwrap())?;
+
+    if symlink_path.exists() {
+        std::fs::remove_file(&symlink_path)?;
+    }
+
+    Ok(())
+}
+
+pub fn remove_symlink(
```

Comment:
Could add the conditional compile, right?

Reply:
I was a bit unsure how to best handle this in rust. With the current implementation I believe this would still need some shims  for the windows case, since otherwise rust complains during compilation that the function is not defined. Maybe there is a straightforward way to statically tell rust that this branch never gets taken on windows though.